### DISCUSSION TITLE
How to Better Implement This Code to Avoid `no-process-exit`?

### DISCUSSION THREAD

**Discussion by manavm1990** [eslint/eslint#14241] [Q&A] [answered]
```javascript
import config from "config";
import { MongoClient } from "mongodb";

const client = new MongoClient(config.mongodbURI, {
  useNewUrlParser: true,
  useUnifiedTopology: true,
});

client.connect();

process.on("SIGINT", () => {
  client.close().then(() => {
    console.info("Closing Mongo client");
  });

  process.exit(0);
});

export default client;
```

W/o going much beyond the issue regarding the linting, basically I am trying to keep my `mongodb` client running 🏃🏾 until such time that we do 'CTRL+C' to interrupt the signal/process deliberately?

<img width="438" alt="Linting error regarding this rule" src="https://user-images.githubusercontent.com/3387791/112081827-5126ab80-8b52-11eb-90f6-a5e9bcd9a80c.png">

**This code does the job.**

But, I don't care for just ignoring 🙈 linter issues unless I have clarity about what it's all about. 

After referencing the [relevant document](https://eslint.org/docs/rules/no-process-exit), I am definitely unsure.

For one thing: 

> It's usually better to throw an error 

What error? I am doing `process.exit(0)` - meaning that I have deliberately and cleanly exited `process`, is it?

Next, WTH is `Process.exit();`?

I tried like this one: `process.exit;`, but ended up having to restart my 💻  b/c it would not allow me to restart `express` on that same port!

For now, I am turning this to `"warn"` - but would like to know...what's it all about? I cannot see why my approach is 'bad' and/or how else I can reliably keep `mongodb` client running consistently and only close when ready. 😖

**Comment by nzakas** [comment, marked as answer]
In this case it’s fine to turn off the rule. In general, process.exit() is best avoided because there’s no way to catch or respond to it. In the entry point to a CLI or service, it makes sense to use it.
